xtext/org.icyphy.linguafranca/src/org/icyphy/generator/GeneratorBase.xtend

```diff
@@ -603,6 +632,205 @@ abstract class GeneratorBase extends AbstractLinguaFrancaValidator {
         }
         result
     }
+    
+    
+    // //////////////////////////////////////////
+    // Protected methods for code generation
+    // of the RTI.
+    
+    // FIXME: Allow target code generators to specify the directory
+    // structure for the generated C RTI?
+    /** Create the runtime infrastructure (RTI) source file.
+     */
+    def generateFederateRTI() {
```

Comment:
I don't think this belongs in `GeneratorBase`. This is all C-specific codegen stuff. If the point is to make it accessible for other code generators perhaps it can remain in `Cgenerator` and be made static or be moved into a separate file. This class is intended to provide the basic framework for generating code for various target languages; there should be no target-specific code in it.